pkg/cmd/server/kubernetes/node.go

```diff
@@ -192,77 +191,77 @@ func (c *NodeConfig) RunSDN() {
 
 // RunProxy starts the proxy
 func (c *NodeConfig) RunProxy() {
-	// initialize kube proxy
-	serviceConfig := pconfig.NewServiceConfig()
-	endpointsConfig := pconfig.NewEndpointsConfig()
-
-	host, _, err := net.SplitHostPort(c.BindAddress)
-	if err != nil {
-		glog.Fatalf("The provided value to bind to must be an ip:port %q", c.BindAddress)
-	}
-	ip := net.ParseIP(host)
-	if ip == nil {
-		glog.Fatalf("The provided value to bind to must be an ip:port: %q", c.BindAddress)
-	}
-
-	protocol := iptables.ProtocolIpv4
-	if ip.To4() == nil {
-		protocol = iptables.ProtocolIpv6
-	}
-
-	syncPeriod, err := time.ParseDuration(c.IPTablesSyncPeriod)
-	if err != nil {
-		glog.Fatalf("Cannot parse the provided ip-tables sync period (%s) : %v", c.IPTablesSyncPeriod, err)
+	protocol := utiliptables.ProtocolIpv4
+	if c.ProxyConfig.BindAddress.To4() == nil {
+		protocol = utiliptables.ProtocolIpv6
 	}
 
 	eventBroadcaster := record.NewBroadcaster()
 	eventBroadcaster.StartRecordingToSink(c.Client.Events(""))
 	recorder := eventBroadcaster.NewRecorder(kapi.EventSource{Component: "kube-proxy", Host: c.KubeletConfig.NodeName})
-	nodeRef := &kapi.ObjectReference{
-		Kind: "Node",
-		Name: c.KubeletConfig.NodeName,
-	}
 
 	exec := kexec.New()
 	dbus := utildbus.New()
-	iptables := iptables.New(exec, dbus, protocol)
+	iptInterface := utiliptables.New(exec, dbus, protocol)
 
 	var proxier proxy.ProxyProvider
 	var endpointsHandler pconfig.EndpointsConfigHandler
-	if /* FIXME */ true {
-		var ipproxier *ipproxy.Proxier
-		ipproxier, err = ipproxy.NewProxier(iptables, exec, syncPeriod, false)
-		proxier = ipproxier
-		endpointsHandler = ipproxier
-	} else {
-		var usproxier *usproxy.Proxier
-		loadBalancer := usproxy.NewLoadBalancerRR()
-		usproxier, err = usproxy.NewProxier(loadBalancer, ip, iptables, util.PortRange{}, syncPeriod, /* FIXME? */ 250 * time.Millisecond)
-		proxier = usproxier
+
+	switch c.ProxyConfig.ProxyMode {
+	case "iptables":
```

Comment:
to work with k8s's default value for ProxyConfig.ProxyMode, this has to be

    case "iptables", "":